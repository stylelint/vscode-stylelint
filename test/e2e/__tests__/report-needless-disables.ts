import { openDocument, waitForDiagnostics, assertDiagnostics, closeAllEditors } from '../helpers';

describe('"stylelint.reportNeedlessDisables" setting', () => {
	afterEach(async () => {
		await closeAllEditors();
	});

	it('should report needless disables when enabled', async () => {
		const { document } = await openDocument('report-disables/needless.css');
		const diagnostics = (await waitForDiagnostics(document)).filter(
			({ code }) => code === '--report-needless-disables',
		);

		assertDiagnostics(diagnostics, [
			{
				code: '--report-needless-disables',
				message: 'Needless disable for "color-hex-length"',
				range: [2, 2, 2, 60],
				severity: 'error',
			},
			{
				code: '--report-needless-disables',
				message: 'Needless disable for "color-hex-length"',
				range: [6, 0, 6, 39],
				severity: 'error',
			},
			{
				code: '--report-needless-disables',
				message: 'Needless disable for "color-hex-length"',
				range: [14, 16, 14, 60],
				severity: 'error',
			},
			{
				code: '--report-needless-disables',
				message: 'Needless disable for "color-hex-length"',
				range: [17, 0, 17, 39],
				severity: 'error',
			},
			{
				code: '--report-needless-disables',
				message: 'Needless disable for "unknown"',
				range: [2, 2, 2, 60],
				severity: 'error',
			},
		]);
	});
});
