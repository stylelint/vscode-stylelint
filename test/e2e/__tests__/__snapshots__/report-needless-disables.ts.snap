// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"stylelint.reportNeedlessDisables" setting should report needless disables when enabled 1`] = `
[
  {
    "code": "--report-needless-disables",
    "message": "Needless disable for "function-name-case"",
    "range": {
      "end": {
        "character": 62,
        "line": 2,
      },
      "start": {
        "character": 2,
        "line": 2,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
  {
    "code": "--report-needless-disables",
    "message": "Needless disable for "function-name-case"",
    "range": {
      "end": {
        "character": 41,
        "line": 6,
      },
      "start": {
        "character": 0,
        "line": 6,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
  {
    "code": "--report-needless-disables",
    "message": "Needless disable for "function-name-case"",
    "range": {
      "end": {
        "character": 62,
        "line": 14,
      },
      "start": {
        "character": 16,
        "line": 14,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
  {
    "code": "--report-needless-disables",
    "message": "Needless disable for "function-name-case"",
    "range": {
      "end": {
        "character": 41,
        "line": 17,
      },
      "start": {
        "character": 0,
        "line": 17,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
  {
    "code": "--report-needless-disables",
    "message": "Needless disable for "unknown"",
    "range": {
      "end": {
        "character": 62,
        "line": 2,
      },
      "start": {
        "character": 2,
        "line": 2,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
]
`;
